components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4890,6 +4890,11 @@ public void addPolicy(Policy policy) throws APIManagementException {
         try {
             if (policy instanceof APIPolicy) {
                 APIPolicy apiPolicy = (APIPolicy) policy;
+                //Check if there's a policy exists before adding the new policy
+                Policy policyIfExists = getAPIPolicy(username, apiPolicy.getPolicyName());
```

Comment:
Shall we use a name like existingPolicy instead of policyIfExists?

Reply:
+1

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/6563/commits/1b840de3a4d81c1894b48b3c040f07574e3b347c